Create typescript type definitions

natekfl opened the pull request.

natekfl committed:
Create index.d.ts

jue89 commented on index.d.ts:
```diff
@@ -0,0 +1,23 @@
+        mode: "netascii" | "octet" | "mail"
+    }
+    export type Response = (buf: Buffer) => void
+    export type Handler = (req: Request, res: Response, next: () => void) => void
```
The `next` callback has an optional parameter that takes an instance of `Error` to abort the handle lookup.

natekfl replied:
Added :)

jue89 reviewed (commented):
Thanks for your PR!

I'll have a deeper look soon. But I already found something to comment on ;)

natekfl committed:
Update index.d.ts

Add err to next(), and change default parameter in createServer to optional as d.ts files don't support default parameters without implementations
